packages/flutter/lib/src/material/drawer_theme.dart

```diff
@@ -0,0 +1,158 @@
+///
+/// Descendant widgets obtain the current [DrawerThemeData] object
+/// using `DrawerTheme.of(context)`. Instances of [DrawerThemeData] can be
+/// customized with [DrawerThemeData.copyWith].
+///
+/// Typically a [DrawerThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.drawerTheme].
+///
+/// All [DrawerThemeData] properties are `null` by default.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class DrawerThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.drawerTheme].
+  const DrawerThemeData({
+    this.backgroundColor,
+    this.scrimColor,
+    this.elevation,
+    this.shape,
+  });
+
+  /// Color to be used for the [Drawer]'s background.
+  final Color? backgroundColor;
+
+  /// Color to be used for the [Drawer]'s scrim when the drawer is open.
+  final Color? scrimColor;
+
+  /// The z-coordinate to be used for the [Drawer]'s elevation.
+  final double? elevation;
+
+  /// Shape to be used for the [Drawer].
+  final ShapeBorder? shape;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  DrawerThemeData copyWith({
+    Color? backgroundColor,
+    Color? scrimColor,
+    double? elevation,
+    ShapeBorder? shape,
+  }) {
+    return DrawerThemeData(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      scrimColor: scrimColor ?? this.scrimColor,
+      elevation: elevation ?? this.elevation,
+      shape: shape ?? this.shape,
+    );
+  }
+
+  /// Linearly interpolate between two drawer themes.
+  ///
+  /// If both arguments are null then null is returned.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static DrawerThemeData? lerp(DrawerThemeData? a, DrawerThemeData? b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    return DrawerThemeData(
+      backgroundColor: Color.lerp(a?.backgroundColor, b?.backgroundColor, t),
+      scrimColor: Color.lerp(a?.scrimColor, b?.scrimColor, t),
+      elevation: lerpDouble(a?.elevation, b?.elevation, t),
+      shape: ShapeBorder.lerp(a?.shape, b?.shape, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      backgroundColor,
+      scrimColor,
+      elevation,
+      shape,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is DrawerThemeData
+        && other.backgroundColor == backgroundColor
+        && other.scrimColor == scrimColor
+        && other.elevation == elevation
+        && other.shape == shape;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('backgroundColor', backgroundColor, defaultValue: null));
```

Comment:
ColorProperty is DiagnosticsProperty<Color> but this property should be DiagnosticsProperty<Color?>. Assuming that's true, we've probably got this error in most uses of ColorProperty in `FooThemeData.debugFillProperties()` definitions.

Same comment about the other properties.

Reply:
Hmm the value field is optional in DiagnosticsProperty, I'm not sure if this is necessary then

Reply:
Sorry, my mistake. The value of `DiagnosticsProperty<T>` is `T?`, so ` DiagnosticsProperty<Color?>` would be nonsensical.